File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) 
    {   
        unordered_map<int , int> val;
        for(auto i :nums)
        {
            val[i]++;
        }
        for(auto i : val)
        {
            if(i.second>1)
            {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        ans=[0]*n
        p=1
        for i in range(n):
            p=p*nums[i]
            ans[i]=p
        p=1
        for i in range(n-1,0,-1):
            ans[i]=ans[i-1]*p
            p=p*nums[i]
        ans[0]=p
        return ans